services/utils/functions.js

```diff
@@ -135,6 +136,15 @@ module.exports = {
       }
     },
 
+    filterOutUnpublished(item) {
+      const relatedItem = item.related && last(item.related);
+      const isHandledByPublshFlow =  relatedItem ? 'published_at' in relatedItem : false;
+      const isRelatedDefinedAndPublished = relatedItem ?
+        isHandledByPublshFlow && get(relatedItem, 'published_at') :
+        false;
+      return item.type === itemType.INTERNAL ? isRelatedDefinedAndPublished  : true
```

Comment:
I think I see some problem here what happens if I have a collection type/single type without draft/publish flow and the item is internal? I think this method will filter that item